docgen/extract/extract.go

```diff
@@ -0,0 +1,78 @@
+package extract
+
+import (
+	"reflect"
+
+	"github.com/golang/protobuf/proto"
+	google_protobuf "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger/options"
+
+	"github.com/gunk/gunk/docgen/pot"
+	"github.com/gunk/gunk/reflectutil"
+)
+
+// Run extract all string from swagger options and store them into pot builder.
+func Run(file *google_protobuf.FileDescriptorProto) (pot.Builder, error) {
+	pb := pot.NewBuilder()
+
+	if proto.HasExtension(file.GetOptions(), options.E_Openapiv2Swagger) {
+		ext, err := proto.GetExtension(file.GetOptions(), options.E_Openapiv2Swagger)
+		if err != nil {
+			return nil, err
+		}
+		pb.AddTranslations(toTranslate(*ext.(*options.Swagger)))
```

Comment:
toTranslate takes an interface, so the type cast here seems unnecessary

Reply:
Yes. I had to add an indirect in the toTranslate func otherwise `NumField` would panic as ext are ptr to swagger struct.

Reply:
Sorry, I misread that code as `(*ext).(*options.Swagger)` :)